core/networkserver/networkserver.go

```diff
@@ -87,10 +89,8 @@ func (n *networkServer) Init(c *component.Component) error {
 	}
 	n.Component.SetStatus(component.StatusHealthy)
 	if n.Component.Monitor != nil {
-		n.monitorStream = n.Component.Monitor.NewNetworkServerStreams(n.Identity.Id, n.AccessToken)
-		go n.Component.Monitor.TickStatus(func() {
-			n.monitorStream.Send(n.GetStatus())
-		})
+		n.monitorStream = n.Component.Monitor.NetworkServerClient(n.Context, grpc.PerRPCCredentials(auth.WithStaticToken(n.AccessToken)))
+		go n.Component.Monitor.TickStatus(func() { n.monitorStream.Send(n.GetStatus()) })
```

Comment:
this goroutine never returns. Which may cause trouble. What if the component is down/disconnected? Do we need to query it's status all the time and send it?